lib/matplotlib/backends/_backend_tk.py

```diff
@@ -22,8 +22,35 @@
     TimerBase, ToolContainerBase, cursors, _Mode, MouseButton,
     CloseEvent, KeyEvent, LocationEvent, MouseEvent, ResizeEvent)
 from matplotlib._pylab_helpers import Gcf
-from . import _tkagg
-from ._tkagg import TK_PHOTO_COMPOSITE_OVERLAY, TK_PHOTO_COMPOSITE_SET
+
+try:
+    from . import _tkagg
+    from ._tkagg import TK_PHOTO_COMPOSITE_OVERLAY, TK_PHOTO_COMPOSITE_SET
+except ImportError as e:
+    # catch incompatibility of python-build-standalone with Tk
+    cause1 = getattr(e, '__cause__', None)
+    cause2 = getattr(cause1, '__cause__', None)
+    if (isinstance(cause1, ImportError) and
+        isinstance(cause2, AttributeError) and
+        "'_tkinter' has no attribute '__file__'" in str(cause2)):
+
+        is_uv_python = "/uv/python" in (os.path.realpath(sys.executable))
+        if is_uv_python:
+            raise ImportError(
+                "Failed to import tkagg backend. You are using a uv-installed python "
+                "executable, which is not compatible with Tk. "
+                "Please update your python via: "
+                "`uv self update && uv python upgrade --reinstall`"
```

Comment:
I think this is fine, but just so you're aware, `uv self update` won't be available for users that have installed uv through their package manager instead of the standalone installer. A more "general" solution would be to "upgrade to the latest version of uv then use `uv python upgrade --reinstall`"